### PR TITLE
Remove goog.addSingletonGetter function

### PR DESCRIPTION
**Pull request by fredj**
See #4128

### CODE REVIEW

File: src/ol/objectutil.js [hunk 1]
```diff
@@ -68,3 +68,21 @@ ol.object.isEmpty = function(object) {
   }
   return !property;
 };
+
+
+/**
+ * Adds a {@code getInstance} static method that always returns the same
```
Comment (tschaub) [inline src/ol/objectutil.js]: Elsewhere we use markdown (backtick) for code formatting.

Comment (fredj) [inline src/ol/objectutil.js]: good catch, fixed

**Comment by tschaub** [comment]
In all cases where `ol.foo.Boo.getInstance()` is called, I think it would be clearer to instead access `ol.foo.bar`.  And in these cases, we never explicitly call `new ol.foo.Bar()`.

In other words, rather than copying the Closure Library style here, I think it would be nicer to get rid of this convention altogether.  So instead of

```js
// where it is defined
goog.addSingletonGetter(ol.style.IconImageCache);

// where it is used
ol.style.IconImageCache.getInstance();
```

We would have this:

```js
// where it is defined
ol.style.iconImageCache = new ol.style.IconImageCache();

// where it is used
ol.style.iconImageCache
```

File: src/ol/objectutil.js [hunk 2]
```diff
@@ -68,3 +68,21 @@ ol.object.isEmpty = function(object) {
...
+
+/**
+ * Adds a `getInstance` static method that always returns the same
+ * instance object.
+ * @param {!Function} object The constructor for the class to add the static
+ *     method to.
+ */
+ol.object.addSingletonGetter = function(object) {
+
+  object.getInstance = function() {
```
Comment (tschaub) [inline src/ol/objectutil.js]: I'm curious if the Compiler knows anything about the type returned by this function.

**Comment by probins** [comment]
status?

**Comment by ahocevar** [comment]
I don't like the `*.instance` naming convention. I like better what @tschaub had suggested, so e.g. 
```js
ol.style.iconImageCache = new ol.style.IconImageCache()
```
instead of
```js
ol.style.IconImageCache.instance = new ol.style.IconImageCache()
```
When addressed, I'd say this is good to merge.

@fredj, are you still active on this, or should we create a new pull request?

**Comment by fredj** [comment]
After changing ```ol.style.IconImageCache.getInstance()``` to ```ol.style.IconImageCache``` the compiler complains that ```goog.require('ol.style.IconImageCache')``` is unnecessary.

What's the right way to fix that?

**Comment by ahocevar** [comment]
I think you'll have to `goog.provide('ol.style.iconImageCache')` instead of `goog.provide('ol.style.IconImageCache' )`. Same goes for `goog.require`.

**Comment by fredj** [comment]
done. I had to make `ol.style.IconImageCache` `@private` to be able to remove the `goog.provide('ol.style.IconImageCache');`

File: src/ol/style/icon.js [hunk 3]
```diff
@@ -638,6 +638,7 @@ ol.style.IconImage_.prototype.unlistenImage_ = function() {
 
 /**
  * @constructor
+ * @private
  */
 ol.style.IconImageCache = function() {
```
Comment (ahocevar) [inline src/ol/style/icon.js]: Now you could rename this to `ol.style.IconImageCache_`

**Comment by ahocevar** [comment]
Looks good now, thanks! Please merge when lights are green.

**Comment by fredj** [comment]
Thanks for the review

**Pull request merged**